bundles/com.amazonaws.eclipse.core/src/com/amazonaws/eclipse/core/mobileanalytics/internal/ToolkitAnalyticsManagerImpl.java

```diff
@@ -45,38 +60,29 @@
     private volatile ToolkitSession currentSession;
 
     /**
-     * @param credentialsProvider
-     *            the credentials provider for Mobile Analytics API calls
-     * @param clientContextConfig
-     *            the client context to be specified in all the PutEvents
-     *            requests.
-     * @throws JsonProcessingException
-     *             if the clientContextConfig fails to be serialized to JSON
-     *             format
+     * @param credentialsProvider the credentials provider for Mobile Analytics API
+     *                            calls
+     * @param clientContextConfig the client context to be specified in all the
+     *                            PutEvents requests.
+     * @throws JsonProcessingException if the clientContextConfig fails to be
+     *                                 serialized to JSON format
      */
-    public ToolkitAnalyticsManagerImpl(
-            AWSCredentialsProvider credentialsProvider,
-            ClientContextConfig clientContextConfig)
-            throws JsonProcessingException {
-
-        this(new MobileAnalyticsBatchClientImpl(
-                credentialsProvider, ClientContextJsonHelper
-                .toJsonString(validateNonNull(clientContextConfig,
-                        "clientContextConfig"))));
+    public ToolkitAnalyticsManagerImpl(AWSCredentialsProvider credentialsProvider, ClientContextConfig clientContextConfig) throws JsonProcessingException {
+
+        this(new MobileAnalyticsBatchClientImpl(credentialsProvider,
```

Comment:
Should MobileAnalyticsBatchClientImpl be renamed, or is it still used a bit?

Reply:
It's still used for error reporting for now. I didn't want to do this rename since it would make a huge diff of rename. We can change it in a later PR if we want.